sql/core/src/test/scala/org/apache/spark/sql/estimation/EstimationSuite.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.estimation
+
+import org.apache.spark.sql.QueryTest
+import org.apache.spark.sql.catalyst.expressions.{Attribute, AttributeMap}
+import org.apache.spark.sql.catalyst.plans.logical._
+import org.apache.spark.sql.catalyst.plans.logical.estimation.EstimationUtils._
+import org.apache.spark.sql.test.SharedSQLContext
+
+
+/**
+ * End-to-end suite testing statistics estimation for logical operators.
+ */
+class EstimationSuite extends QueryTest with SharedSQLContext {
+  import testImplicits._
+
+  /** Table info: all estimation tests are conducted on these tables. */
+  private val estimationTestData = Seq(
+    ("estimation_test1", Seq("key11", "key12"), Seq[(Int, Int)]((1, 10), (2, 10))))
+
+  /** Original column stats */
+  val colStatForKey11 = ColumnStat(2, Some(1), Some(2), 0, 4, 4)
+  val colStatForKey12 = ColumnStat(1, Some(10), Some(10), 0, 4, 4)
+
+  override def beforeAll(): Unit = {
```

Comment:
in order to write a unit test, can we create a logical plan node with a some fake statistics that's passed in? that way we don't need everything end to end and can even put this in the catalyst package.